pkg/media/samplebuilder/samplebuilder_test.go

```diff
@@ -223,6 +223,27 @@ func TestSampleBuilder(t *testing.T) {
 			maxLate:          50,
 			maxLateTimestamp: 2000,
 		},
+		{
+			// shamelessly stolen from webrtc-rs
+			message: "Padding",
+			packets: []*rtp.Packet{
+				{Header: rtp.Header{SequenceNumber: 5000, Timestamp: 1}, Payload: []byte{1}},
+				{Header: rtp.Header{SequenceNumber: 5001, Timestamp: 1}, Payload: []byte{2}},
+				{Header: rtp.Header{SequenceNumber: 5002, Timestamp: 1, Marker: true}, Payload: []byte{3}},
+				{Header: rtp.Header{SequenceNumber: 5003, Timestamp: 1}, Payload: []byte{}},
+				{Header: rtp.Header{SequenceNumber: 5004, Timestamp: 1}, Payload: []byte{}},
+				{Header: rtp.Header{SequenceNumber: 5005, Timestamp: 3}, Payload: []byte{1}},
```

Comment:
It's worth having a test that also has `Marker: true` here, i.e. a packet that follows a run of padding packets and is both a start and end of a partition i.e. a very small video partition that fits inside 1200 bytes. This is something I observed with Chrome and it's covered in the webrtc-rs code.

Reply:
@k0nserv would that be this one ?
https://github.com/webrtc-rs/webrtc/pull/309/files#diff-0e05f283d453fbc18b4b99c7bf67c1ccd8e630a44e57047ee7a2a5fcc70dfbccR1055

Reply:
That's testing that it can built a sample from such a packet in isolation. This one test such a packet following a padding run https://github.com/webrtc-rs/webrtc/pull/309/files#diff-0e05f283d453fbc18b4b99c7bf67c1ccd8e630a44e57047ee7a2a5fcc70dfbccR1099 where "Valid packet 3"  is both start and end of a partition

Reply:
I've added a test for this with some caveats:
- for some reason setting `withHeadChecker: true` crashes on trying to index 0 of an empty array, I'm guessing it's due to missing padding handling?
- likewise there is no `PrevPaddingPackets` field yet
- and the expected results are not the same as in webrtc-rs, presumably again because padding isn't yet handled here